### ISSUE TITLE
Accessibility - Screens are Missing Headings

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Ilab-do-good#27]
Screens are missing heading which screen reader users them to move around the screen to find content.

Once I started using the application I noticed what should be headings seemed to be un-ordered lists (OL) with one item in them which is issue #8.

Please see the following screens as an example of screens that need headings.

- What are Child and Forced Labor
- Why Develop a Social Compliance System?
- The Basics of a Social Compliance System

I'm sure I will be adding more to the list as I go along.

**Comment by adriennerlong** [comment]
The developers are aware of this problem and are making it a high priority to address before 9/11.

**Comment by jfc3-dol** [comment]
@adriennerlong,

Thanks, for getting the developers to fix this item. It's important for those using screen readers.

**Comment by jfc3-dol** [comment]
Thanks, for making the content a heading.

The only issue is that you shouldn't skip from heading level one (h1) to a heading level four (h4). Most if not all of the headings should be a heading level two (h2) unless some of the items structurally fall under the heading level two (h2) then they should be a heading level three (h3). You wouldn't skip chapters in a book. You would want to go from chapter one to five and then back to two.

**Comment by adriennerlong** [comment]
ICF, please incorporate section headings h1-4 according to the guidance sent 9/1 pm

**Comment by jfc3-dol** [comment]
@adriennerlong,

Headings have been added but on the "what are Child Labor and Forced Labor?" screen the heading level skips from a heading level one (h1) to a heading level three (h3).

The text "Child Labor" and "Forced Labor" should be heading level two (h2)  but are not read as such. They are read as the following for both of them "Expand content toggle button double press to toggle the setting." I would expect VoiceOver to read the text as a heading level two (h2) item and then mention it's able to expand etc. not sure if that is going to work that way in iOS or not.

**Comment by jfc3-dol** [comment]
For Android some headings have been added.

The following two screens have a heading level one (h1) but the other items that should be heading level two (h2) items are being used as toggle buttons that have no text being read to tell what they are. They should be heading level two (h2) items as well as the toggle button.
 
- What are Child and Forced Labor
- The Basics of a Social Compliance System

**Comment by adriennerlong** [comment]
From ICF: Partially fixed.  Headers need to be read out as labels

**Comment by jfc3-dol** [comment]
VoiceOver reads the following two screens heading level one (h1) but the other items that should be heading level two (h2) items. They are being used as toggle buttons that have no text being read to tell what they are. They should be heading level two (h2) items as well as the toggle button.
 
- What are Child and Forced Labor
- The Basics of a Social Compliance System

VoiceOver reds that they can be toggled open and closed but don't announce what section/heading level two (h2) item it is used for.

**Comment by jfc3-dol** [comment]
TlakBack reads the following two screens heading level one (h1) but the other items that should be heading level two (h2) items. They are being used as toggle buttons that have no text being read to tell what they are. They should be heading level two (h2) items as well as the toggle button.

    What are Child and Forced Labor
    The Basics of a Social Compliance System

TlakBack reads that they can be toggled open and closed but don't announce what section/heading level two (h2) item it is used for and that it's heading level two.

**Comment by sanganalakshay** [comment]
Fixed

**Comment by jfc3-dol** [comment]
TalkBack reads that they can be toggled open and closed but don't announce what section/heading level two (h2) item it is used for and that it's heading level two.

It's nice that TalkBack tells a user the item can be toggled, but it's not helpful if the application does not tell me what I'm opening up. The issues still are not fixed.

**Comment by sanganalakshay** [comment]
@jfc3-dol 
On my android device on "What are Child and Forced Labor" page, the talkback reads "Child Labor Heading level two click to expand content double tap to toggle" for Child labor and similarly for Forced Labro
Is this what you are referring to?

**Comment by jfc3-dol** [comment]
@sanganalakshay,

On the "What are Child and Forced Labor" screen my version of Android is not reading the "Child Labor Heading level two" part. All I'm getting is the "click to expand content double tap to toggle" part.

Let me go back and attempt to do this again once I'm done working my way through the iOS issues I'm testing now.

The iOS version is working as you described. VoiceOver is reading "Child Labor Heading level two click to expand content double tap to toggle" and similarly for the other heading level two (h2) headings.

Removing the iOS label since the issue is corrected.

**Comment by sanganalakshay** [comment]
@jfc3-dol  Could you please confirm that you are running the latest version of the android apk that I posted in my last commit on 28 Nov at 9.29 am? 
Because I am running that same apk on my device and it is reading fine.

Thanks,
Akshay

**Comment by giridhar** [comment]
@sanganalakshay its the latest version that john is testing. I have installed it from him.

**Comment by jfc3-dol** [comment]
On both of our Android devices on the "What are Child and Forced Labor" screen TalkBack is not reading the "Child Labor Heading level two" part. All I'm getting is the "click to expand content double tap to toggle" part.

**Comment by sanganalakshay** [comment]
@jfc3-dol  
could you please try again with the new apk I have pushed out today morning... its working on my device by installing that apk

**Comment by jfc3-dol** [comment]
@sanganalakshay,

I will get Giri to load it on my devices when he has time to do so.

**Comment by jfc3-dol** [comment]
@sanganalakshay,

Thanks, now both our devices are now reading the heading level two text and that they can be toggled.

Removing the Android Label and closing the issue.
